timer.js

```diff
@@ -25,48 +20,170 @@ function startTimer(durationSeconds = 25 * 60) {
+      document.getElementById("timerSeconds").textContent = seconds;
+      document.getElementById("timer").classList.add("timer-active");
+      // save the current remaining time to state
+      state.timer.remainingTimeOnResume = timerDurationS;
+    
+      timerDurationS -= 1;
+      // handle when the timer runs out
+      if (timerDurationS < 0) {
+        clearInterval(state.timerIntervalId);
+        if (state.isWorkInterval) {
+          startTimer(300); // Start rest interval
+          state.isWorkInterval = false;       
+        } else {
+          startTimer(1500); // Start work interval
+          state.isWorkInterval = true;
+        }
+      }
+    }, 1 * 1000);
+  } else {
+    alert("Chose todo from table");
+    return;
+  }
+  
+} 
 
 function pauseTimer() {
   clearInterval(state.timerIntervalId);
   document.getElementById("timer").classList.remove("timer-active"); // Remove class when paused
+  updateCurrentTomato();
 }
 
 function stopTimer() {
-  //updateTomatoCount();
   clearInterval(state.timerIntervalId);
+  state.isWorkInterval = !state.isWorkInterval; // Toggle to next interval
+  state.timer.remainingTimeOnResume = undefined; // Reset remaining time
+  state.currentId = undefined;
+  state.currentTodo = undefined;
+
+  if (state.isWorkInterval) {
+    timerDurationS = 25 * 60;
+  } else {
+    timerDurationS = 5 * 60;
+  }
+
+  minutes = parseInt(timerDurationS / 60, 10); // 25
+  seconds = parseInt(timerDurationS % 60, 10); // 0
+
+  minutes = minutes < 10 ? "0" + minutes : minutes;
+  seconds = seconds < 10 ? "0" + seconds : seconds;
+
+  // updating the UI with the state
+  document.getElementById("timerMinutes").textContent = minutes;
+  document.getElementById("timerSeconds").textContent = seconds;
   document.getElementById("timer").classList.remove("timer-active"); // Remove class when stopped
-  // showTodo.textContent = "";
 
-  //if (currentTodoCell) {
-  //  currentTodoCell.textContent = document.getElementById('timerMinutes').textContent + ":" + document.getElementById('timerSeconds').textContent;
-  //}
-  state.isWorkInterval = !state.isWorkInterval; // Toggle to next interval
+  updateCurrentTomato();
+  updateCurrentTodo();
+}
+// save the current tomato to the state and reset the current tomato
+function updateCurrentTomato() {
+  if (state.currentTomato) {
+    state.timer.tomatoes.push(state.currentTomato);
+    state.currentTomato = undefined;
+    updateTable(); 
+  }
+}
+
+function currentTomatoStartEnd() { 
+  now = new Date();
+  currentMonth = now.getMonth();
+  currentDay = now.getDate();
+  currentYear = now.getFullYear();
+  currentHour = now.getHours();
+  currentMinute = now.getMinutes();
+  currentSecond = now.getSeconds();
 
-  startTimer(state.isWorkInterval ? 1500 : 300);
+  if (state.isWorkInterval) {
+    if (!state.currentTomato) {
+      state.currentTomato = {
+        id: state.currentId,
+        start: {
+          month: currentMonth,
```

Comment:
I see 3 repetitions of this thing

```
{
          month: currentMonth,
          day: currentDay,
          year: currentYear,
          hour: currentHour,
          minute: currentMinute,
          second: currentSecond,
        }
```

could be cleaner to put that into a constant and reuse it ? ( https://kentcdodds.com/blog/aha-programming not DRY, not WET, but AHA )

Reply:
I made like that. AI helped me. 
![image](https://github.com/antondurinin/ToDo/assets/167963627/7d0f5329-206f-4a22-9eb0-f2bb0011b733)
Very interesting AHA, i will try it. Some times this abstractions can really crazy for mind, i try to evade them. 
While I am solving your TODO, i am rebuilding more and more code beyond this TODO